### ISSUE TITLE
memory leak detection

### ISSUE THREAD

**Issue by rhp7rholype** [geany/geany#3682]
Hello, I installed the Geany program and then launched it. After I left for several hours, I came back and the computer began to freeze, I looked at the memory, it displays 7GB occupied by Geany. I opened the statistics and displayed a diagram of continuous memory growth. All plugins were disabled and then removed. Tested on versions 1.38 and 2.0

Screen after 1 min

![2023-11-11 12_54_11-Свойства_ geany exe (8476)](https://github.com/geany/geany/assets/64293069/189e4a85-8df4-43e7-9f54-ba168ba98366)

**Comment by rhp7rholype** [comment]
after 10 min

![2023-11-11 13_19_33-Свойства_ geany exe (8568)](https://github.com/geany/geany/assets/64293069/faa1ebf1-71df-4b00-b1d8-f496f941f6c7)

**Comment by elextr** [comment]
What are the three sections of the graph?

**Comment by rhp7rholype** [comment]
CPU
Memory (Closed)
IO

**Comment by elextr** [comment]
Have you set auto `reload_clean_doc_on_file_change` and opened a continually changing file, like a log file?  If you have `keep_edit_history_on_reload` the history of reloads will accumulate and will use up memory.

**Comment by rhp7rholype** [comment]
Yeah, there was an op keep_edit_history_on_reload. I disable this option.
Reloaded the Geany program.
While I write a message, my memory has already increased to 150 MB, and it's still growing

**Comment by rhp7rholype** [comment]
![2023-11-11 15_12_42-Настройки](https://github.com/geany/geany/assets/64293069/ac409fb4-9d82-4488-b3f9-c45056802fe7)

**Comment by rhp7rholype** [comment]
files.show_keep_edit_history_on_reload_msg is unchecked too

**Comment by elextr** [comment]
What OS?

**Comment by rhp7rholype** [comment]
windows 10
![2023-11-11 15_18_53-Windows_ сведения](https://github.com/geany/geany/assets/64293069/cf336651-7140-4e08-9f1f-a80f786ecb63)

**Comment by elextr** [comment]
Don't know on Windows, on Linux would tell you to run `geany -c /tmp/non_existant` to make Geany run with a default configuration, but don't know if you can do that on Windows.

Are you sure _all_ plugins are disabled, not ticked in the plugin manager?

**Comment by rhp7rholype** [comment]
I tryed reinstall Geany 1.38, 2.0...
I Checked yesterday and today
Result, memory leak

**Comment by elextr** [comment]
What is triggering Geany to execute? Geany should do nothing unless something (eg you typing) triggers it.  AFAICT no other Windows user has complained about this, but maybe someone has seen it.

**Comment by rhp7rholype** [comment]
I don’t know, I just run it through Explorer.
It will be necessary to check in a couple of days using leak detection programs

**Comment by rhp7rholype** [comment]
on other PC normal stable 24MB
Maybe problem in library...

**Comment by b4n** [comment]
If you have the opportunity tu run a leak detector on this it might be interesting, although given the info here I'm wondering if it's not something outside geany that interferes (any kind of program that hook into others). Or one of the support libraries, maybe GTK has a bad leak in the drawing code on some specific situations?
